lib/auth/grpcserver.go

```diff
@@ -5204,6 +5204,65 @@ func (g *GRPCServer) GetHeadlessAuthentication(ctx context.Context, req *proto.G
 	return authReq, trace.Wrap(err)
 }
 
+// WatchPendingHeadlessAuthentications watches the backend for pending headless authentication requests for the user.
+func (g *GRPCServer) WatchPendingHeadlessAuthentications(_ *emptypb.Empty, stream proto.AuthService_WatchPendingHeadlessAuthenticationsServer) error {
+	auth, err := g.authenticate(stream.Context())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	filter := types.HeadlessAuthenticationFilter{
+		Username: auth.context.User.GetName(),
```

Comment:
What happens if Nop (aka a client with no client cert) tries to hit this endpoint? Do we error out at `g.authenticate`?

Reply:
Yes, `g.authenticate` would error out here:
```
func (a *authorizer) Authorize(ctx context.Context) (*Context, error) {
	if ctx == nil {
		return nil, trace.AccessDenied("missing authentication context")
	}
	userI, err := UserFromContext(ctx)
	if err != nil {
		return nil, trace.Wrap(err)
	}
```

Reply:
`g.authenticate` returns a `ServerWithRoles` with the authentication context mapped to `Nop`; everything seems to eventually fail because some operation is prohibited to non-local-users, but we should make that explicit at the beginning of every one of the headless authentication functions in `auth_with_roles`, IMO.